### PR TITLE
many: simpler snap.File, disentangle squashfs (no more snap import), direct registration

### PR DESCRIPTION
**Pull request by pedronis**

### CODE REVIEW

File: snap/file.go [hunk 1]
```diff
@@ -23,21 +23,20 @@ import (
 	"bytes"
 	"fmt"
 	"os"
+
+	"github.com/ubuntu-core/snappy/snap/squashfs" // to register it
```
Comment (zyga) [inline snap/file.go]: The comment is confusing. It's not that the import is doing some interesting side-effects any more. I'd drop the comment.

File: asserts/toolbelt/main.go [hunk 2]
```diff
@@ -1,204 +0,0 @@
-// -*- Mode: Go; indent-tabs-mode: t -*-
```
Comment (mvo5) [inline asserts/toolbelt/main.go]: Is this file removal on purpose in this particular branch? it looks a bit out-of-place here, especially given that the description does not mention it.

**Comment by mvo5** [comment]
Looks fine, just one question.

**Comment by zyga** [comment]
+1, with one minor inline comment

File: asserts/toolbelt/main.go (hunk 2 above)
Comment (pedronis) [inline asserts/toolbelt/main.go]: well the other changes break it, and it was never advertized (it's really demoware that so far compiled but what already conceptually broken)

**Comment by pedronis** [comment]
retest this please

**Comment by niemeyer** [comment]
Very nice, thanks!

**Pull request merged**